src/MyEventEmitter.js

```diff
@@ -1,12 +1,116 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+const { checkForErrors } = require('./checkForErrors');
+
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    checkForErrors(eventName, listener);
+
+    const { events } = this;
+
+    if (events[eventName]) {
+      events[eventName].push(listener);
+    } else {
+      events[eventName] = [listener];
+    }
+
+    return this;
+  }
+
+  once(eventName, listener) {
+    checkForErrors(eventName, listener);
+
+    const { events } = this;
+
+    const onceListener = (...args) => {
+      listener(...args);
+      this.off(eventName, onceListener);
+    };
+
+    if (events[eventName]) {
+      events[eventName].push(onceListener);
+    } else {
+      events[eventName] = [onceListener];
+    }
+
+    return this;
+  }
+
+  off(eventName, listener) {
+    checkForErrors(eventName, listener);
+
+    const { events } = this;
+
+    events[eventName] = events[eventName].filter(func => func !== listener);
+
+    return this;
+  }
+
+  emit(eventName, ...args) {
+    checkForErrors(eventName, () => {});
```

Comment:
I think it is better to use default value for parameter `listener` instead of passing an empty function